### ISSUE TITLE
Not enough memory on image.c Buffer

### ISSUE THREAD

**Issue by MoritzBrueckner** [armory3d/armorcore#52]
The following error occurs in some Armory projects with many textures when running with Armorcore:

```
Not enough memory on image.c Buffer.
```

It seems to be caused by Kinc trying to allocate textures on a pre-allocated buffer which isn't large enough. Previously, there always was an additional commit in armory3d/Kinc along the lines of "use std malloc for image.h" to instead use standard allocation, which seems to be missing in the currently used version of Kinc. After adding it back the error is gone:

```cpp
// Kinc/Sources/kinc/image.h line 216

#ifndef KINC_IMAGE_STANDARD_MALLOC
#define STBI_MALLOC(sz) buffer_malloc(sz)
#define STBI_REALLOC(p, newsz) buffer_realloc(p, newsz)
#define STBI_FREE(p) buffer_free(p)
#endif
```
[Permalink to the Kinc sources](https://github.com/Kode/Kinc/blob/82e7f4619bbc4e34b02d76c13ff9cabb70511c03/Sources/kinc/image.h#L216-L218)

Some additional questions:
1. I'm not sure whether the compiler is allowed to optimize the buffer declaration away if it's not used, since that's not part of the `#ifndef KINC_IMAGE_STANDARD_MALLOC` block, will it always allocate memory even if it's unused?
2. Would it make sense to only use standard allocation if the buffer is full and else use the buffer first? This could improve performance to some extent.
3. What do you think of creating a repository that contains patches for Armory's versions of Kha and Kinc so that it's possible to contribute to them (in case of things that don't belong into the Kode versions, like [here](https://github.com/armory3d/armorcore/issues/37))? Would that technically be possible or is it difficult because of Kha/Kinc changing lines and files all the time?

**Comment by luboslenco** [comment]
Will try to look into https://github.com/Kode/Kinc/issues/407 since it's been 3+ years, to fix this for once and all. Once that is done let's investigate https://github.com/armory3d/armorcore/issues/37.

**Comment by e2002e** [comment]
Hi, just to let you know. This issue happened to me too, and the version of Kink in the sdk from armory3d repo is already fixed as you shown.

**Comment by luboslenco** [comment]
This should be (finally) resolved now at the https://github.com/armory3d/armorcore/tree/armsdk branch.

Basically it uses a custom image load function, not the kinc one:
https://github.com/armory3d/armorcore/blob/armsdk/Sources/main.cpp#L994
